### ISSUE TITLE
About L2 Regularize

### ISSUE THREAD

**Issue by makefile** [Yagami123/Caffe-AdamW-AdamWR#1]
Shouldn't we disable the L2 Regularize or set another weight_decay param for AdamW?
In function `SGDSolver<Dtype>::ApplyUpdate()` there calls `Regularize` which also use `weight_decay` param. So, I think maybe it will be conflict with the weight decay in AdamW.

**Comment by Yagami123** [comment]
@makefile SGDSolver call Normalize(), Regularize(), ComputeUpdateValue() in function `SGDSolver<Dtype>::ApplyUpdate()`  in order. Adamwr and other solvers extend SGDSolver

**Comment by Yagami123** [comment]
@makefile SGDSolver call Normalize(), Regularize(), ComputeUpdateValue() in function SGDSolver<Dtype>::ApplyUpdate() in order. Adamwr and other solvers extend SGDSolver
which will not change function `Regularize()`. It just redefinition function `ComputeUpdateValue()`. 
So you can set L1 Regularize or others by yourself.

**Comment by makefile** [comment]
Yes, but I wonder whether it will get good result if use L1/L2 together with AdamW weight decay.In other words, when use AdamW do we still need add regularization?

**Comment by Yagami123** [comment]
Of course we need, `AdamW` is just to revise `Regularize` of implementation of Adam's L1/L2 which is not to redefine a new regularize method. Refers to paper https://arxiv.org/pdf/1711.05101.pdf, reading section I and II.

**Comment by makefile** [comment]
Thanks. The paper makes me much more clear.
